sembast/lib/src/api/filter.dart

```diff
@@ -82,3 +86,16 @@ abstract class Filter {
   factory Filter.custom(bool Function(RecordSnapshot record) matches) =>
       SembastCustomFilter(matches);
 }
+
+/// Provides convenience methods for combining multiple [Filter]s.
+extension FilterCombination on Filter {
```

Comment:
This makes `FilterCombination` public. i.e. it meens some people could do `import 'sembast.dart' show FilterCombination` at some point to maintain forever. I want to avoid polluting a generic namespace more than I do now so I'd rather have it prefixed for now with a name such as `SembastFilterCombination`